os/arch/arm/src/s5j/s5j_i2s.c

```diff
@@ -297,9 +297,9 @@ static uint32_t i2s_samplerate(struct i2s_dev_s *dev, uint32_t rate);
 
 static uint32_t i2s_txdatawidth(struct i2s_dev_s *dev, int bits);
 static int i2s_send(struct i2s_dev_s *dev, struct ap_buffer_s *apb, i2s_callback_t callback, void *arg, uint32_t timeout);
-static int i2s_stop(struct i2s_dev_s *dev);
-static int i2s_pause(struct i2s_dev_s *dev);
-static int i2s_resume(struct i2s_dev_s *dev);
+static int i2s_stop(struct i2s_dev_s *dev, i2s_ch_dir_t dir);
```

Comment:
Would it be possibel to stop only RX or TX if HOST is master?

Reply:
Similar logic is applied in IRQ handler. 
So, I suppose this must work. 

Also, if stopping either RX or TX would completely stop the I2S clock, then why do we have separate register setting to stop the complete I2S core?

Reply:
@kishore-sn Is it common for all chipset? (My knowledge is limited on other chipsets)
If you are sure that these changes are portable on Cypress board also, then I can dismiss the hold. 
But, if you are NOT sure, then better to wait for Ivan's agreement.

Reply:
@kishore-sn please dont scope only for ARTIK. Please check if this will work on others as well.

Reply:
I cannot be sure about how I2S works in all SOC's. In my opinion, there has to be a way to individually control the RX and TX channels. I have checked the I2S implementation in bcm43909 and also in an old android project based on Samsung SOC. They seem to have similar type of implementation. For example, checkout the below API's from bcm code:

static int i2s_stop_stream( platform_i2s_port_t port, platform_i2s_direction_t dir );

So, I suggest that we go ahead with this implementation for now. When we work on a new SOC, we would either need to find some way of implementing the similar feature or add a work around in our code. 

@Taejun-Kwon If you have any suggestions, please let me know.

Reply:
I checked many i2s codes and I found seperate rx & tx with direction is common style.(I jsut checked android & zephyr's some h/w)
But I think benix's opinion also very important.
Let's discuss about it more. I wil invite you guys.